File: 8 - Dictionaries and Maps/Solution.rb
n = gets.chomp.to_i
hashMap= {}
for x in 0..n-1
    arr=gets.chomp.split(" ")
    hashMap[arr[0]]=arr[1]
end
while true
    query = gets
    break unless query
    name = query.chomp
    if hashMap.key?(name)
        puts "#{name}=#{hashMap[name]}"
    else
        puts "Not found"    
    end
end

File: 8 - Dictionaries and Maps/Solution.go
package main

import "fmt"

func main() {
	var n int
	m := make(map[string]int)

	fmt.Scan(&n)

	var name string
	var number int
	for i := 0; i < n; i++ {
		fmt.Scan(&name)
		fmt.Scan(&number)
		m[name] = number
	}

	var query string
	for {
		_, err := fmt.Scanf("%s", &query)
		if err != nil {
			break
		}
		if value, ok := m[query]; ok {
			fmt.Printf("%s=%d\n", query, value)
		} else {
			fmt.Println("Not found")
		}
	}

}
